### DISCUSSION TITLE
Why is espressoBuildConfig needed as desired capability when using espresso?

### DISCUSSION THREAD

**Discussion by attrideepak** [appium/appium#16325] [Q&A] [answered]
My espressoBuildConfig is something like this:
`{   "additionalAndroidTestDependencies": [     "androidx.core:core:1.5.0",     "androidx.activity:activity:1.2.0-beta01"   ],   "additionalAppDependencies": [     "org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.5.0"  ] }`

My app is using an n number of dependencies but why only specifying these few dependencies is sufficient to run the test?

**Comment by KazuCocoa** [comment, marked as answer]
It depends on your app and espresso (instrumentation) stuff. (This _espresso_ is Android's espresso framework)

Appium espresso server has https://github.com/appium/appium-espresso-driver/blob/master/espresso-server/app/build.gradle.kts#L87 by default. So in case an app is enough with them, it does not require additional dependencies.
I assume some of your dependencies do not need espresso (instrumentation) process, so you do not need them.

I don't know recent espresso documentation, but you can learn more about Andoird espresso themselves probably around https://developer.android.com/training/testing/espresso
